localstack-core/localstack/utils/xray/trace_header.py

```diff
@@ -0,0 +1,139 @@
+# Source: https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/trace_header.py
```

Comment:
@thrau Do we have some best practices around re-using open source code, especially from AWS sources?

Reply:
we don't, but i think what you're doing here is fine! maybe we could be a bit more verbose:

```
# This file is part of LocalStack.
# It is adapted from aws-xray-sdk-python licensed under the Apache License 2.0.
# You may obtain a copy of the Apache License 2.0 at http://www.apache.org/licenses/LICENSE-2.0
# Original source: https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/trace_header.py
# Modifications: Add optional lineage field for https://docs.aws.amazon.com/lambda/latest/dg/invocation-recursion.html
```

Reply:
Done 👍